src/Servers/Kestrel/Core/src/Internal/Http3/Http3Connection.cs

```diff
@@ -530,8 +544,14 @@ private void UpdateStreamTimeouts(long timestamp)
                     await outboundControlStreamTask;
                 }
 
+                // Use graceful close reason if it has been set.
+                if (reason == ConnectionEndReason.Unknown && _gracefulCloseReason != ConnectionEndReason.Unknown)
```

Comment:
Nit: I don't think `&& _gracefulCloseReason != ConnectionEndReason.Unknown` actually accomplishes anything.

Reply:
What do you mean? Is it that `_gracefulCloseReason != ConnectionEndReason.Unknown` condition could be removed and an unknown value is set to unknown?

Reply:
It's been a while, but I probably meant that `reason == ConnectionEndReason.Unknown` implies `_gracefulCloseReason != ConnectionEndReason.Unknown` at this point in the code.  I may well have been mistaken.